packager/media/formats/mp2t/ts_segmenter.cc

```diff
@@ -102,7 +102,7 @@ Status TsSegmenter::AddSample(const MediaSample& sample) {
   if (sample.is_encrypted())
     ts_writer_->SignalEncrypted();
 
-  if (!ts_writer_file_opened_ && !sample.is_key_frame())
+  if (!ts_writer_buffer_initialized_ && !sample.is_key_frame())
```

Comment:
Thought about it again. Can we rename it to start_of_new_segment_, which is more appropriate?